### PR TITLE
Remove space after icon if no text is present, fix projecthydra/sufia…

### PR DESCRIPTION
**Pull request by hackmastera**
…#1354

### CODE REVIEW

File: app/helpers/sufia/sufia_helper_behavior.rb [hunk 1]
```diff
@@ -114,8 +114,10 @@ def link_to_field(fieldname, fieldvalue, displayvalue = nil)
     end
 
     def iconify_auto_link(text, showLink = true)
+      # this block is only executed when a link is inserted;
+      # if we pass text containing no links, it just returns text.
       auto_link(text) do |value|
-        "<i class='glyphicon glyphicon-new-window'></i>&nbsp;#{value if showLink}<br />"
+        "<i class='glyphicon glyphicon-new-window'></i>#{"&nbsp;#{value}" if showLink}<br />"
```
Comment (mbklein) [inline app/helpers/sufia/sufia_helper_behavior.rb]: Nested `#{}`s almost never work. I think what you need here is something like `#{'(&nbsp;' + value) if showLink}`

**Comment by jcoyne** [comment]
How about a test for this method?

File: app/helpers/sufia/sufia_helper_behavior.rb (hunk 1 above)
Comment (hackmastera) [inline app/helpers/sufia/sufia_helper_behavior.rb]: Hm, works fine when I look at it in the internal-app. but maybe it's confusing a parser or compiler; will try your suggestion.

**Comment by hackmastera** [comment]
@mbklein thanks for the suggestion; unfortunately no effect. It's nicer to look at your way, though.

@awead says this will be resolved by https://github.com/projecthydra/sufia/pull/1357

@jcoyne I would maybe write a test if i could get tests passing for sufia but it's a losing battle for me right now and I'm demoralized. If that is a blocker y'all can close the PR as will not merge.

**Comment by mjgiarlo** [comment]
@hackmastera #1357 is merged. Could you rebase your PR? I'd love to get this merged.

**Comment by hackmastera** [comment]
@mjgiarlo it's ready

**Pull request merged**